lib/full-site-editing/block-templates.php

```diff
@@ -450,47 +456,6 @@ function gutenberg_filter_wp_template_unique_post_slug( $override_slug, $slug, $
 		$override_slug = $slug;
 	}
 
-	// Template slugs must be unique within the same theme.
-	// TODO - Figure out how to update this to work for a multi-theme
-	// environment.  Unfortunately using `get_the_terms` for the 'wp-theme'
-	// term does not work in the case of new entities since is too early in
-	// the process to have been saved to the entity.  So for now we use the
-	// currently activated theme for creation.
-	$theme = wp_get_theme()->get_stylesheet();
-	$terms = get_the_terms( $post_ID, 'wp_theme' );
-	if ( $terms && ! is_wp_error( $terms ) ) {
-		$theme = $terms[0]->name;
-	}
-
-	$check_query_args = array(
-		'post_name__in'  => array( $override_slug ),
-		'post_type'      => $post_type,
-		'posts_per_page' => 1,
-		'no_found_rows'  => true,
-		'post__not_in'   => array( $post_ID ),
-		'tax_query'      => array(
-			array(
-				'taxonomy' => 'wp_theme',
-				'field'    => 'name',
-				'terms'    => $theme,
-			),
-		),
-	);
-	$check_query      = new WP_Query( $check_query_args );
-	$posts            = $check_query->get_posts();
-
-	if ( count( $posts ) > 0 ) {
-		$suffix = 2;
-		do {
-			$query_args                  = $check_query_args;
-			$alt_post_name               = _truncate_post_slug( $override_slug, 200 - ( strlen( $suffix ) + 1 ) ) . "-$suffix";
-			$query_args['post_name__in'] = array( $alt_post_name );
-			$query                       = new WP_Query( $query_args );
-			$suffix++;
-		} while ( count( $query->get_posts() ) > 0 );
-		$override_slug = $alt_post_name;
-	}
-
 	return $override_slug;
 }
 add_filter( 'pre_wp_unique_post_slug', 'gutenberg_filter_wp_template_unique_post_slug', 10, 5 );
```

Comment:
I think we can remove the entire filter now, no?

Reply:
I don't think so, because it still needs to change it from the default behaviour, just in a different way to how it was doing before.

Reply:
Hmm, are we changing it from the default behaviour though? I thought that the remnants are just boilerplate really...

Reply:
It is changing it from the default behaviour. If you look at the apply_filters call in context, you can see it's one of those filters that triggers an early return if it gets a non-null value.

Reply:
Yeah, it does that, which means bypassing WP's own "unique slug" logic. My point is that that made sense when we had our own logic to ensure that, but now that we don't, it doesn't seem to make sense to continue to bypass it.

Reply:
I believe this filter is still needed because the theme mod is saved based on the post slug, and this is checked on the save_post action hook, by which time the slug would have already had the suffix appended to it.

Reply:
It can now be removed as of 764d484